### PR TITLE
211 support for showing all tests in curent workspace and picking which tests to run in vs code extension

### PR DESCRIPTION
**Pull request by oakmount66**

### CODE REVIEW

File: vs-code-extension/client/src/extension.ts [hunk 1]
```diff
@@ -0,0 +1,407 @@
+// The module 'vscode' contains the VS Code extensibility API
+// Import the module and reference it with the alias vscode in your code below
+
+//Include files ved byg/udgivelse?
+//Dele cut og Cobol Check extension op?
+//Klasser i typescript virker ikke?
```
Comment (Rune-Christensen) [inline vs-code-extension/client/src/extension.ts:6]: Please remove comments in danish.

File: vs code extension/Cobol-check/bin/cobol-check-0.2.8.jar
Comment (Rune-Christensen) [inline vs code extension/Cobol-check/bin/cobol-check-0.2.8.jar:1]: Do we not need the 0.2.8 file in the bin directory?
The extension.ts file points to this location.

File: vs-code-extension/client/src/extension.ts [hunk 2]
```diff
@@ -0,0 +1,407 @@
...
+	var prevFile: vscode.TestItem = file
+	var tmpDir = rootUri
+	var tmpFile = null
+	var tmpData = null
+
+	for(var i =1;i<dirArr.length;i++){
+		tmpDir = tmpDir + '/' + dirArr[i]
+		const existing = prevFile.children.get(tmpDir);
+		
+		if(!existing){
```
Comment (Rune-Christensen) [inline vs-code-extension/client/src/extension.ts:276]: Please reverse this if else statement.
So that we avoid the negation in the if statement.

**Review by Rune-Christensen — CHANGES_REQUESTED** [review]
Looks good overall, a couple of small issues that I would like fixed before merging.

File: vs-code-extension/Cobol-check/output/testResults.html
[the remark this answers is not in the document]
Comment (oakmount66) [reply vs-code-extension/Cobol-check/output/testResults.html:1]: Done

File: vs-code-extension/client/src/Helpers/ExtensionHelper.ts [hunk 3]
```diff
@@ -15,13 +15,23 @@ export async function handleCobolCheckOut(output : CobParser.CobolCheckOutputPar
...
 			case CobParser.OutputState.NoIssues:
+				console.log("NO ISSUES")
 				testResultFile = appendPath(vsCodeInstallDir, await getConfigurationValueFor(configPath, 'test.results.file'));
 				htmlResult = await getTextFromFile(testResultFile + '.html');
 				showWebWiev('Test Results - ' + getCurrentProgramName(), htmlResult)
 				resolve(true);
 				break;
 
+			case CobParser.OutputState.TestFailed:
+				console.log("TEST FAILES")
```
[the remark this answers is not in the document]
Comment (oakmount66) [reply vs-code-extension/client/src/Helpers/ExtensionHelper.ts:26]: Done

File: vs-code-extension/client/src/services/CobolCheckInputParser.ts [hunk 4]
```diff
@@ -0,0 +1,37 @@
+import * as vscode from 'vscode';
+
+const testCaseRegex = /^\s*(TESTCASE)\s*(.*)/i;
+// const testRegex = /^\s*(Expect)\s*(.*)\s*(not?)\s(>|<|to\s*be)\s*(.*)\s*/i;
```
[the remark this answers is not in the document]
Comment (oakmount66) [reply vs-code-extension/client/src/services/CobolCheckInputParser.ts:4]: Done

File: vs-code-extension/client/src/extension.ts (hunk 1 above)
Comment (oakmount66) [reply vs-code-extension/client/src/extension.ts:6]: Done

File: vs-code-extension/client/src/extension.ts (hunk 2 above)
Comment (oakmount66) [reply vs-code-extension/client/src/extension.ts:276]: Done

File: vs code extension/Cobol-check/bin/cobol-check-0.2.8.jar
Comment (oakmount66) [reply vs code extension/Cobol-check/bin/cobol-check-0.2.8.jar:1]: The jar file is created by the gradle command "./gradlew prepareDistribution", so it's up to the developer to create it.

**Review by Rune-Christensen — APPROVED** [review]

**Review by Rune-Christensen — APPROVED** [review]

**Pull request merged**